insights/parsers/numa_cpus.py

```diff
@@ -0,0 +1,84 @@
+"""
+NUMACpus - file ``/sys/devices/system/node/node[0-9]*/cpulist``
+===============================================================
+
+This parser will parse the content from cpulist file, from individual
+NUMA nodes. This parser will return data in (dict) format.
+
+Sample Content from ``/sys/devices/system/node/node0/cpulist``::
+
+    0-6,14-20
+
+Examples:
+    >>> type(numa_cpus_obj)
+    <class 'insights.parsers.numa_cpus.NUMACpus'>
+    >>> numa_cpus_obj.numa_node_name
+    'node0'
+    >>> numa_cpus_obj.numa_node_details() == {'numa_node_range': ['0-6', '14-20'], 'total_cpus': 14, 'numa_node_name': 'node0'}
+    True
+    >>> numa_cpus_obj.numa_node_cpus
+    ['0-6', '14-20']
+    >>> numa_cpus_obj.total_numa_node_cpus
+    14
+"""
+
+from insights import Parser, parser
+from insights.specs import Specs
+
+
+@parser(Specs.numa_cpus)
+class NUMACpus(Parser):
+    """
+    Parse `/sys/devices/system/node/node[0-9]*/cpulist` file, return a dict
+    which contains total number of CPUs per numa node.
+    """
+
+    def __init__(self, context):
+        self.data = {}
+        self._cpu_ranges = []
+        self.numa_node = context.path.rsplit("/")[-2] if context.path else None
```

Comment:
This assignment can also be shifted to the `parser_content`, since the `self.file_path` is the `context.path`.  Then the `__init__` can be saved.